target/bin/delmailuser

```diff
@@ -1,92 +1,108 @@
+if ! ${MAILDEL}
+then
+  read -r -p "Do you want to delete the mailbox as well (removing all mails) ? [Y/n] " MAILDEL_CHOSEN
+  [[ ${MAILDEL_CHOSEN} =~ (y|Y| ) ]] && MAILDEL=true
+fi
+
 (
   flock -e 200
 
   for USER in "${@}"
   do
     # very simple plausibility check
-    [[ ${USER} != *"@"*"."* ]] && errex "No valid address: ${USER}"
+    [[ ${USER} != *'@'*'.'* ]] && errex "No valid address: ${USER}"
 
     declare -a MAILARR
     MAILARR[0]="${USER%@*}"
     MAILARR[1]="${USER#*@}"
 
-    # XXX ${USER} must not contain /s and other syntactic characters
+    # ${USER} must not contain /s and other syntactic characters
     USER=$(escape "${USER}")
 
-    if ! sed -i "/^""${USER}""|/d" "${DATABASE}"
+    if [[ -f ${DATABASE} ]]
     then
-      errex "${USER} couldn't be deleted in ${DATABASE}. ${?}"
+      if ! sed -i "/^${USER}|/d" "${DATABASE}"
+      then
+        errex "${USER} couldn't be deleted in ${DATABASE}. ${?}"
+      fi
     fi
 
-    # Delete all aliases where the user is the only recipient( " ${USER$}" )
-    # Delete user only for all aliases that deliver to multiple recipients ( ",${USER}" "${USER,}" )
-    if sed -i -e "/ ""${USER}""$/d" \
-        -e "s/,""${USER}""//g" \
-        -e "s/""${USER}"",//g" "${ALIAS_DATABASE}"
+    if [[ -f ${ALIAS_DATABASE} ]]
     then
-      echo "${USER} and potential aliases deleted." || errex "Aliases for ${USER} couldn't be deleted in ${ALIAS_DATABASE}. ${?}"
+      # delete all aliases where the user is the only recipient( " ${USER$}" )
+      # delete user only for all aliases that deliver to multiple recipients ( ",${USER}" "${USER,}" )
+      if sed -i \
+        -e "/ ${USER}$/d" -e "s/,${USER}//g" -e "s/${USER},//g" \
+        "${ALIAS_DATABASE}"
+      then
+        echo "${USER} and potential aliases deleted."
+      else
+        errex "Aliases for ${USER} couldn't be deleted in ${ALIAS_DATABASE}."
+      fi
     fi
 
     # remove quota directives
     if [[ -f ${QUOTA_DATABASE} ]]
     then
-      sed -i -e "/^${USER}:.*$/d" "${QUOTA_DATABASE}" || errex "Quota for ${USER} couldn't be deleted in ${QUOTA_DATABASE}. ${?}"
+      if ! sed -i -e "/^${USER}:.*$/d" "${QUOTA_DATABASE}"
+      then
+        errex "Quota for ${USER} couldn't be deleted in ${QUOTA_DATABASE}."
+      fi
     fi
 
-    if [[ ${MAILDEL} != "y" ]]
+    if ! ${MAILDEL}
     then
-      read -r -p "Do you want to delete the mailbox as well(all mails will be removed)?(y/n) " MAILDEL
-      echo
+      echo "Leaving the mailbox untouched. If you want to delete it at a later point use 'sudo docker exec mail rm -R /var/mail/${MAILARR[1]}/${MAILARR[0]}'"
+      exit 0
     fi
 
-    [[ ${MAILDEL} != "y" ]] && errex "Leaving the mailbox untouched. If you want to delete it at a later point use \"sudo docker exec mail rm -R /var/mail/${MAILARR[1]}/${MAILARR[0]}\""
-
-    if rm -r -f "/var/mail/${MAILARR[1]}/${MAILARR[0]}"
+    if [[ -d "/var/mail/${MAILARR[1]}/${MAILARR[0]}" ]]
```

Comment:
Will hardcoding `/var/mail/` work with setups that have a custom/different maildir set? Thinking about the PR we merged a few days ago (#1792)

Reply:
This is most likely going to be a concern in the future - I think this PR shouldn't worry about this for now:)